src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -27,6 +29,14 @@ def __init__(self, config):
         BaseWorkerThread.__init__(self)
         self.config = config
         self.drainAPI = DrainStatusAPI()
+        self.condorAPI = PyCondorAPI()
+
+        if hasattr(config, "Tier0Feeder"):
+            self.reqAuxDB = None
+            self.speedDrainConfig = None
```

Comment:
I think this `self.speedDrainConfig` line can be set above, outside the if-else block.
Then in the algorithm method, just below the "Checking agent drain status" log, we add something like
```
if self.reqAuxDB:
    self.checkSpeedDrainThresholds()
```